migrator/src/main/java/com/google/phonenumbers/migrator/MetadataReader.java

```diff
@@ -0,0 +1,74 @@
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.i18n.phonenumbers.metadata.model.RangesTableSchema;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * A class used to represent a given metadata zip file. Each {@link MetadataReader} instance
+ * represents a zip file containing Csv Tables that can be imported based on specified
+ * region codes without the need to extract the whole zip file.
+ */
+public final class MetadataReader {
+  private final ZipFile metadataZipFile;
+
+  public MetadataReader(String fileLocation) throws IOException {
+    this.metadataZipFile = new ZipFile(fileLocation);
+  }
+
+  /**
+   * Returns an Optional {@link CsvTable} correlating to a given regionCode. If no ranges.csv file
+   * can be found in the {@link #metadataZipFile} for the region code, an empty Optional is returned
+   */
+  public Optional<CsvTable<RangeKey>> getCsvTable(String regionCode) throws IOException {
+    String CsvTableLocation = "metadata/" + regionCode + "/ranges.csv";
```

Comment:
[nit/style] Prefer 'lowerCamelCase' for variable names (i.e. "csvTableLocation"). Did IntelliJ have a warning about this in the code? I would hope so.

If you get used to ensuring that the right-hand-side vertical bar in IntelliJ is clear of warnings and errors, it lets you spot and fix these little things as you go. However IntelliJ has some "opinions" about code which Google doesn't share, so you might end up deciding to turn some of the inspections off (we can go over the inspections stuff during a bit of pair programming if you want).